tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot

```diff
@@ -1,26 +1,120 @@
 *** Settings ***
-Resource    ../../../Resources/ODS.robot
+Resource            ../../../Resources/ODS.robot
+Test Setup          Begin Metrics Web Test
+Test Teardown       End Metrics Web Test
+
 
 *** Variables ***
+@{RECORD_GROUPS}    Availability Metrics    SLOs - JupyterHub    SLOs - ODH Dashboard    SLOs - RHODS Operator    SLOs - Traefik Proxy    Usage Metrics
 
-@{RECORD_GROUPS}  Availability Metrics  SLOs - JupyterHub  SLOs - ODH Dashboard  SLOs - RHODS Operator  SLOs - Traefik Proxy  Usage Metrics
+@{ALERT_GROUPS}     Builds                  DeadManSnitch    RHODS-PVC-Usage    SLOs-haproxy_backend_http_responses_total    SLOs-probe_success
 
-@{ALERT_GROUPS}  Builds  DeadManSnitch  RHODS-PVC-Usage  SLOs-haproxy_backend_http_responses_total  SLOs-probe_success
 
 *** Test Cases ***
-
 Test Existence of Prometheus Alerting Rules
-  [Tags]  Smoke  Sanity  ODS-509
-  Check Prometheus Alerting Rules
+    [Tags]    Smoke
+    ...       Sanity
+    ...       ODS-509
+    Check Prometheus Alerting Rules
 
 Test Existence of Prometheus Recording Rules
-  [Tags]  Smoke  Sanity  ODS-510
-  Check Prometheus Recording Rules
+    [Tags]    Smoke
+    ...       Sanity
+    ...       ODS-510
+    Check Prometheus Recording Rules
+
+Test Metric "Notebook CPU Usage" On ODS Prometheus
+    [Documentation]    Verifing the notebook cpu usage showing on RHODS promethues
+    [Tags]    Sanity
+    ...       Tier1
+    ...       ODS-178
+    ${cpu_usage_before} =    Read Current CPU Usage
+    Run Jupyter Notebook For 5 Minutes
+    ${cpu_usage_after} =    Read Current CPU Usage
+    Should Not Be Equal    ${cpu_usage_before}    ${cpu_usage_after}
+
+Test Metric "Rhods_Total_Users" On ODS Prometheus
+    [Documentation]    Verifies that metrics rhods_total_users and jupyterhub_total_users have the same value in Prometheus
+    [Tags]    Sanity
+    ...       Tier1
+    ...       ODS-628
+    ${expression} =    Set Variable    rhods_total_users&step=1  #step = 1 beacuase it will return value of current second
+    ${rhods_total_users} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    ${expression}
+    ${expression} =    Set Variable    jupyterhub_total_users&step=1  #step = 1 beacuase it will return value of current second
+    ${jupyterhub_total_users} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}
+    ...    ${expression}
+    Should Be Equal    ${rhods_total_users.json()["data"]["result"][0]["value"][-1]}
```

Comment:
should both the metrics always have the same values? If the last values are equal, it doesn't imply there isn't no point in the time where the metrics have different values. I would check that along all the considered time period, the two metrics  reports the same total number of users.

Reply:
yes
accordingly to Testcase in polarion they should same

Reply:
The values should be the same (jupyterhub informs prometheus periodically of this value). I don't think it's necessary to check the historic values to compare

Reply:
i'm not 100% conviced about this..but you feel sure about it, okay